### PR TITLE
add e2e test for InterpretStatus

### PR DESCRIPTION
**Pull request by zhuwint**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Add e2e test for InterpretStatus in resource interpreter.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

NONE

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Thanks~

/assign

File: examples/customresourceinterpreter/webhook/app/workloadwebhook.go [hunk 1]
```diff
@@ -123,3 +126,22 @@ func (e *workloadInterpreter) responseWithExploreInterpretHealth(workload *workl
...
+
+	marshaledBytes, err := json.Marshal(status)
+	if err != nil {
+		return interpreter.Errored(http.StatusInternalServerError, err)
+	}
+
+	res := interpreter.Succeeded("")
+	res.RawStatus = &runtime.RawExtension{
+		Raw:    marshaledBytes,
+		Object: &workloadv1alpha1.Workload{},
```
Comment (XiShanYongYe-Chang) [inline examples/customresourceinterpreter/webhook/app/workloadwebhook.go:143]: This line seems can be removed.

File: test/e2e/resourceinterpreter_test.go [hunk 2]
```diff
@@ -186,4 +186,21 @@ var _ = ginkgo.Describe("Resource interpreter webhook testing", func() {
...
+				gomega.Expect(clusterDynamicClient).ShouldNot(gomega.BeNil())
+
+				memberWorkload := framework.GetWorkload(clusterDynamicClient, workloadNamespace, workloadName)
+				memberWorkload.Status.ReadyReplicas = *workload.Spec.Replicas
+				framework.UpdateWorkload(clusterDynamicClient, memberWorkload, cluster, "status")
+
+				klog.Infof("Waiting for workload(%s/%s) on cluster %s collecting correctly status", workloadNamespace, workloadName, cluster)
+				currentWorkLoad := framework.GetWorkload(clusterDynamicClient, workloadNamespace, workloadName)
+				gomega.Expect(currentWorkLoad.Status.ReadyReplicas).Should(gomega.Equal(*workload.Spec.Replicas))
+			}
```
Comment (XiShanYongYe-Chang) [inline test/e2e/resourceinterpreter_test.go:203]: Shouldn't we check the object of `Work`?

Comment (zhuwint) [reply test/e2e/resourceinterpreter_test.go:203]: Is that we should check the whole WorkLoad.Status object ?

File: test/e2e/resourceinterpreter_test.go [hunk 3]
```diff
@@ -186,4 +187,46 @@ var _ = ginkgo.Describe("Resource interpreter webhook testing", func() {
...
+				memberWorkload.Status.ReadyReplicas = *workload.Spec.Replicas
+				framework.UpdateWorkload(clusterDynamicClient, memberWorkload, cluster, "status")
+
+				workName := names.GenerateWorkName(workload.Kind, workload.Name, workload.Namespace)
+				workNamespace, err := names.GenerateExecutionSpaceName(cluster)
+				gomega.Expect(err).Should(gomega.BeNil())
+
+				gomega.Eventually(func(g gomega.Gomega) bool {
+					work, err := karmadaClient.WorkV1alpha1().Works(workNamespace).Get(context.TODO(), workName, metav1.GetOptions{})
+					if err != nil {
```
Comment (XiShanYongYe-Chang) [inline test/e2e/resourceinterpreter_test.go:207]: For err case, we can exit with `g.Expect(err).NotTo(gomega.HaveOccurred())`, so we don't have to wait until it times out.

**Comment by XiShanYongYe-Chang** [comment]
Thanks~
/lgtm

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**